heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/filter/helper/DBMongoImpl.java

```
@@ -294,12 +295,14 @@ private MongoDatabase database() {
 
      private <T> Object getValueId(T object) {
 
-          Field id = Arrays.stream(object.getClass().getDeclaredFields()).filter(field -> field.getAnnotation(Id.class) != null).findFirst().get();
-          id.setAccessible(true);
-          try {
-               return id.get(object);
-          } catch (IllegalArgumentException | IllegalAccessException e) {
-               log.error(e.getMessage(), e);
+          Optional<Field> idOptional = Arrays.stream(object.getClass().getDeclaredFields()).filter(field -> field.getAnnotation(Id.class) != null).findFirst();
+          Field id;
+          if(idOptional.isPresent()){
+               id = idOptional.get();
+               id.setAccessible(true);
+               try {
+                    return id.get(object);
+               } catch (IllegalArgumentException | IllegalAccessException ignored) { }
```

Why you are ignoring the exception?

---

Reverted